XCUITests/SearchTest.swift

```diff
@@ -218,4 +218,38 @@ class SearchTests: BaseTestCase {
         waitForExistence(app.textFields["url"], timeout: 20)
         waitForValueContains(app.textFields["url"], value: "google")
     }
+    
+    func testSearchIconOnAboutHome() {
+        waitForTabsButton()
+        navigator.performAction(Action.OpenNewTabFromTabTray)
+        waitForTabsButton()
+        
+        // Search icon is displayed.
+        waitForExistence(app.buttons["TabToolbar.stopReloadButton"])
+        XCTAssertEqual(app.buttons["TabToolbar.stopReloadButton"].label, "Search")
+        XCTAssertTrue(app.buttons["TabToolbar.stopReloadButton"].exists)
+        app.buttons["TabToolbar.stopReloadButton"].tap()
```

Comment:
This could be added as an action to the screenGraph like:
`screenState.gesture(forAction: Action.OpenSearchBarFromSearchButton, transitionTo: URLBarOpen) {userState in
            app.buttons["TabToolbar.stopReloadButton"].tap()
        }`

and so you can do that in the test as:
`navigator.performAction(Action.OpenSearchBarFromSearchButton)`

Once you do that, you are in the [UrlBarOpen](https://github.com/mozilla-mobile/firefox-ios/blob/main/XCUITests/FxScreenGraph.swift#L405) state, so to type the url you could use:
`navigator.performAction(Action.LoadURLByTyping)`
to type the url. 

In this case the url is the one by default used by the graph: 
https://github.com/mozilla-mobile/firefox-ios/blob/main/XCUITests/FxScreenGraph.swift#L261

Reply:
oh, I forgot about those.  would I add this `screenState.gesture` into an existing map.addScreenState?  or would I need to create a new one?

Reply:
Ah forgot to mention that. You can add that gesture to the exisiting HomePanelsScreen screenState. The gestures/actions are added to the different screenStates and in general there is a screenState per screen, similar to the robots approach